beacon_node/eth1/src/service.rs

```diff
@@ -53,127 +50,12 @@ const CACHE_FACTOR: u64 = 2;
-        );
-        let state = endpoint_state(&endpoint.client, &self.config_chain_id, &self.log).await;
-        *value = Some(state.clone());
-        if state.is_err() {
-            crate::metrics::inc_counter_vec(
-                &crate::metrics::ENDPOINT_ERRORS,
-                &[&endpoint.client.to_string()],
-            );
-            crate::metrics::set_gauge(&metrics::ETH1_CONNECTED, 0);
-        } else {
-            crate::metrics::set_gauge(&metrics::ETH1_CONNECTED, 1);
-        }
-        state
-    }
-
-    /// Return the first successful result along with number of previous errors encountered
-    /// or all the errors encountered if every none of the fallback endpoints return required output.
-    pub async fn first_success<'a, F, O, R>(
-        &'a self,
-        func: F,
-    ) -> Result<(O, usize), FallbackError<SingleEndpointError>>
-    where
-        F: Fn(&'a HttpJsonRpc) -> R,
-        R: Future<Output = Result<O, SingleEndpointError>>,
-    {
-        let func = &func;
-        self.fallback
-            .first_success(|endpoint| async move {
-                match self.state(endpoint).await {
-                    Ok(()) => {
-                        let endpoint_str = &endpoint.client.to_string();
-                        crate::metrics::inc_counter_vec(
-                            &crate::metrics::ENDPOINT_REQUESTS,
-                            &[endpoint_str],
-                        );
-                        match func(&endpoint.client).await {
-                            Ok(t) => Ok(t),
-                            Err(t) => {
-                                crate::metrics::inc_counter_vec(
-                                    &crate::metrics::ENDPOINT_ERRORS,
-                                    &[endpoint_str],
-                                );
-                                if let SingleEndpointError::EndpointError(e) = &t {
-                                    *endpoint.state.write().await = Some(Err(e.clone()));
-                                } else {
-                                    // A non-`EndpointError` error occurred, so reset the state.
-                                    reset_endpoint_state(endpoint).await;
-                                }
-                                Err(t)
-                            }
-                        }
-                    }
-                    Err(e) => Err(SingleEndpointError::EndpointError(e)),
-                }
-            })
-            .await
-    }
-
-    pub async fn reset_errorred_endpoints(&self) {
-        for endpoint in &self.fallback.servers {
-            if let Some(state) = get_state(endpoint).await {
-                if state.is_err() {
-                    reset_endpoint_state(endpoint).await;
-                }
-            }
-        }
-    }
-}
-
 /// Returns `Ok` if the endpoint is usable, i.e. is reachable and has a correct network id and
 /// chain id. Otherwise it returns `Err`.
 async fn endpoint_state(
```

Comment:
There's another instance of `"trying fallbacks"` in this function. We should delete all of them.